t/porting/bidi-control.t

```diff
@@ -0,0 +1,61 @@
+#!/usr/bin/perl
+use v5.20.0;
+use warnings;
+
+use utf8 ();
+
+BEGIN {
+  @INC = '..' if -f '../TestInit.pm';
+  require './test.pl';
+}
+
+use TestInit qw(T); # T is chdir to the top level
+
+find_git_or_skip();
+
+my $cdup = `git rev-parse --show-cdup`;
+die "couldn't git-rev-parse --show-cdup\n" if $?;
+chomp $cdup;
+
+if (length $cdup) {
+  chdir $cdup or die "can't chdir to git root: $!";
+}
+
+my @files = `git ls-files`;
+die "couldn't git-ls-files\n" if $?;
+
+chomp @files;
+
+if (0) {
+  # Is this a useful filter?  Should we test every file?  Do we contain binary
+  # files, etc? -- rjbs, 2021-11-02
+  my %is_interesting = map {; $_ => 1 } qw( bat c h pl pm sh t xs );
+  @files = grep {; /\.([a-z]+)\z/ && $is_interesting{$1} } @files;
+}
```

Comment:
My inclination is that it shouldn't be useful: our source code repository should contain source. But I don't think we've ever had an explicit policy in that regard, so we might need to create one. And I don't know what's in the tree today (but see below).

In particular, I think it should be possible to have files with no extension, and it should be ok to test those as well, but a quick check shows we don't have such today. What we do appear to have is 127 different extensions - 89 if you ignore `README.\w+` - and it feels like it would be safer to exclude some extensions from this check (if we really must) than to list only those that should be included. In earlier discussions I suggested to start without exceptions, and add a mechanism for them only at the point we find we really need one.

Peeking at some of the rarer extensions, I note that `os2/diff.configure` has been an empty file since 2014. Several other outliers look to be support files for tests (eg for `IO::Compress`), but a bunch more sure look like code.